search/constants.py

```diff
@@ -0,0 +1,5 @@
+from typing import List
+
+GATEWAY_URL: str = ""
+ELASTICSEARCH_HOSTS: List[str] = []
```

Comment:
Still not a fan of these 'defaults'. Presumably empty values means it's broken after the fact (ie rather than not starting up, it does start then fails if you try to do anything).

Reply:
Empty values would mean it breaks when trying to send anything to Elasticsearch as the client will not be able to connect. However, due to the added logging, this will not break. It will simply log the errors and not persist anything (silently failing).

What would be your desired approach?

Reply:
Presumably not having the file means the app won't start? That seems the best approach to me (it should be documented that you need to add it).

If you add in the wrong values etc, then logged failures are fine.

Reply:
What if I leave it in as an example but comment everything out so the values will not be loaded?

Reply:
After much consideration. Leaving the default variables with empty values but I have added a check that is run on application start up that will raise an exception if required settings are not set.

Reply:
Now application refuses to start if settings are missing:
```
[15:02:45][giorgio@Newton:~/code/libero-search]$ docker run -it libero/search:master flask run
 * Serving Flask app "search"
 * Environment: production
   WARNING: Do not use the development server in a production environment.
   Use a production WSGI server instead.
 * Debug mode: off
Traceback (most recent call last):
  File "/usr/local/bin/flask", line 10, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/site-packages/flask/cli.py", line 894, in main
    cli.main(args=args, prog_name=name)
  File "/usr/local/lib/python3.7/site-packages/flask/cli.py", line 557, in main
    return super(FlaskGroup, self).main(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/flask/cli.py", line 767, in run_command
    app = DispatchingApp(info.load_app, use_eager_loading=eager_loading)
  File "/usr/local/lib/python3.7/site-packages/flask/cli.py", line 293, in __init__
    self._load_unlocked()
  File "/usr/local/lib/python3.7/site-packages/flask/cli.py", line 317, in _load_unlocked
    self._app = rv = self.loader()
  File "/usr/local/lib/python3.7/site-packages/flask/cli.py", line 372, in load_app
    app = locate_app(self, import_name, name)
  File "/usr/local/lib/python3.7/site-packages/flask/cli.py", line 254, in locate_app
    return find_best_app(script_info, module)
  File "/usr/local/lib/python3.7/site-packages/flask/cli.py", line 76, in find_best_app
    app = call_factory(script_info, app_factory)
  File "/usr/local/lib/python3.7/site-packages/flask/cli.py", line 116, in call_factory
    return app_factory()
  File "/srv/app/search/__init__.py", line 13, in create_app
    check_required_settings()
  File "/srv/app/search/utils.py", line 16, in check_required_settings
    raise RequiredSettingNotSet(message)
search.exceptions.RequiredSettingNotSet: GATEWAY_URL is required and must be set in search.constants.py
```
Probably the error message has a reference to an outdated file name?

Reply:
This is the expected behaviour and it's the correct file.

Reply:
Ah it's a typo, `search/constants.py` vs `search.constants.py`

Reply:
updated.